### ISSUE TITLE
Airship test and some improvement suggestions

### ISSUE THREAD

**Issue by by003** [vcmi/vcmi#4051]
**Describe your proposal**
![image](https://github.com/vcmi-mods/horn-of-the-abyss/assets/127420617/1985ed42-947f-4b8e-b542-0f34a87dc2ce)

I use core:boat to create airship, with this codes:
```
{
	"core:boat" : {
		"types" : {
			"airshipFactory" : {
				"removable": true,
				"rmg" : {
				},
				"layer" : "air",
				// If set to true, it is possible to attack wandering monsters from boat
				"onboardAssaultAllowed" : false,
				// If set to true, it is possible to visit object (e.g. pick up resources) from boat
				"onboardVisitAllowed" : false,
				"actualAnimation" : "factory/map/objects/AAir00.def",
				"overlayAnimation" : "ABM03_.def",
				"flagAnimations" : [
				"factory/map/objects/AAirF00L.def",
				"factory/map/objects/AAirF00G.def",
				"factory/map/objects/AAirF00R.def",
				"factory/map/objects/AAirF00D.def",
				"factory/map/objects/AAirF00B.def",
				"factory/map/objects/AAirF00P.def", 
				"factory/map/objects/AAirF00W.def",
				"factory/map/objects/AAirF00K.def"],
				"templates" : {
					"default" : {
						"animation" : "factory/map/objects/AAir00_E.def",
						"visitableFrom" : [ "+++", "+-+", "+++" ],
						"mask" : [ "VVV","VVV","VVV", "VAV" ]
					}
				}
			}
		}
	}
}
```
Now there are some problems:
1. Hero cannot enter the water directly, but can directly go ashore.
![image](https://github.com/vcmi-mods/horn-of-the-abyss/assets/127420617/887d2fd2-cd3d-496b-a3c6-eb3921165feb)
2. If hero needs to enter the water, a boat is needed to become a ‘springboard’, if there is no boat, this hero can never enter the water.
![image](https://github.com/vcmi-mods/horn-of-the-abyss/assets/127420617/3bd698c1-57ee-4c93-99f4-14043d71bac6)
3. This hero can attack others or pick theresources, depends on "onboardAssaultAllowed" and "onboardVisitAllowed" are set on 'true'.
![image](https://github.com/vcmi-mods/horn-of-the-abyss/assets/127420617/ea3e02b4-6a4a-4090-9805-ab74983a0480)
4. This hero CANNOT get off the ship, need a button to help him.

**Comment by by003** [comment]
Same in [https://github.com/vcmi-mods/horn-of-the-abyss/issues/107](url)

**Comment by by003** [comment]
Airship yard can be placed, but it can only build airship near the water, it cannot be used in land.
![image](https://github.com/vcmi/vcmi/assets/127420617/ce36b728-f498-4618-a644-8839c953dc29)

**Comment by by003** [comment]
@DjWarmonger Have you plan to make airship and other hota 1.7 abilities in next version?

**Comment by DjWarmonger** [comment]
Not really, it's a rarely used feature. Getting Factory abilities to work has a priority.

**Comment by by003** [comment]
> Getting Factory abilities to work has a priority.

I plan to update factory mod for VCMI in June. I hope detonation skill can be actived first, and recruit 8 creatures should be available.

**Comment by GeorgeK1ng** [comment]
@mrhaandi Maybe you are interested in this one?

**Comment by by003** [comment]
@mrhaandi If you are interested, the main problems with airships at present are:

1. They cannot be manufactured in the airship yard.
2. Even if you board an airship, you cannot disembark using a button.

Furthermore, I hope HotA's eagle eye can be used as well, just like scholar, but only applies to the enemy.

It would be great if we could complete more HotA features!

**Comment by mrhaandi** [comment]
I have looked at the code and this one will need some work.
Basically, there are two ways:
1. implement airship disembark as teleport
   - easy, but ad hoc
   - own logic
   - no pathfinding
   - ai will get into trouble
2. extend the pathfinding algorithm to understand air - land transitions
   - the `MoveHero` logic (client and server) needs to understand whether an air - air transition or an air -land transition is done (currently, this information is lost)
   - ai should be able to use air ships similar to boats
   - uniform logic with sailing

I'll see whether option 2. is manageable. In any case, UI needs to be adapted to differentiate: fly to destination vs disembark at destination.
Edit: pathfinder flying mechanics reflect the fly spell and not airship travel (and there is no logic to distinguish the two).

Note to myself: map for airship tests: [airship.zip](https://github.com/user-attachments/files/24169306/airship.zip)

**Comment by Alucard648** [comment]
Another suggestion: objects that can only be visited while riding airship, i.e. treasuries on floating islands.

**Comment by by003** [comment]
If it is completed and has the same function as HotA, you can close it.
@GeorgeK1ng

**Comment by by003** [comment]
Most of the airship issues have been resolved.

A few remain:
1. The airship dock recruitment interface should be static; currently, it only shows an animation of rising and falling.
<img width="339" height="210" alt="Image" src="https://github.com/user-attachments/assets/1da0499f-4362-4f81-a305-b3def66dccb5" />

2. The rising and falling animation loops continuously during flight, which is visually unappealing.

https://github.com/user-attachments/assets/6170ea12-2d6a-43f2-9cbc-e21228e01ada

3. In HotA's setting, airships cannot attack any neutral forces or enemies, but it seems that battles will occur when they get close.
4. Airships should not play the sound of horses' hooves when sailing on land.
